modules/siddhi-core/src/main/java/org/wso2/siddhi/core/util/ExecutionPlanRuntimeBuilder.java

```diff
@@ -88,9 +88,7 @@ public ExecutionPlanRuntimeBuilder(ExecutionPlanContext executionPlanContext) {
     public void defineStream(StreamDefinition streamDefinition) {
         DefinitionParserHelper.validateDefinition(streamDefinition, streamDefinitionMap, tableDefinitionMap,
                 windowDefinitionMap);
-        if (!streamDefinitionMap.containsKey(streamDefinition.getId())) {
-            streamDefinitionMap.putIfAbsent(streamDefinition.getId(), streamDefinition);
-        }
+        streamDefinitionMap.put(streamDefinition.getId(), streamDefinition);
```

Comment:
No, we should not do this. In the first definition we might add annotation but in the second one, we might not have an annotation as its an inferred definition. So at least need to have putIfAbsent()